src/SAWScript/Crucible/MIR/Builtins.hs

```diff
@@ -1089,18 +1054,156 @@ setupCrucibleContext rm =
      -- TODO! there's a lot of options setup we need to replicate
      --  from SAWScript.Crucible.LLVM.Builtins
 
+     -- There is quite a bit of faff below, all for the sake of translating
+     -- top-level static values. See Note [Translating MIR statics in SAW] for
+     -- a high-level description of what this code is doing.
```

Comment:
The code below here is the implementation of `Note [Translating MIR statics in SAW]`.